pkg/sharky/shard.go

```diff
@@ -128,7 +128,11 @@ LOOP:
 			if writes != nil {
 				sh.slots.wg.Add(1) // Done after the slots process pops from slots.in
 				go func() {
-					sh.slots.in <- slot
+					defer sh.slots.wg.Done()
+					select {
+					case sh.slots.in <- slot:
+					case <-sh.quit:
```

Comment:
There may be a possibility to have slots not read from this channel. I think that there is still need some improvement around insurance that all slots are pushed before quit.

Reply:
I have added another wg for the "limbo" free slots writes to wait for them after the quit is closed. That way the original waitgroup is easier to follow but also ensures that no free slots are dropped.